### ISSUE TITLE
15 seconds ago 🚀 ML Engineer Build AI Content Performance Predictor & AB Testing System for Social Media SaaS

### ISSUE THREAD

**Issue by abdullahqureshi6** [wahab-o16labs/UW-Submitted-Bids#49]
Summary
We’re Intuity Sync, an AI-driven content automation platform that helps creators, agencies, and brands generate and schedule high-performing content across TikTok, LinkedIn, Instagram, X, and Facebook.

We’ve built the infrastructure. Now we want to go next level by integrating machine learning into our product, helping users automatically learn what content works best and why.

We’re looking for a sharp, product-minded ML engineer to build our first content prediction and auto-optimization system.

🧠 The Problem We’re Solving
Creators waste weeks AB testing content,  trying different captions, post times, and image styles, just to figure out what resonates with their audience. We want to replace that trial-and-error with AI-driven prediction and automatic improvement.

✅ What You’ll Be Building
This is greenfield, high-impact work. Your mission is to implement the first version of our Content Performance Prediction + AB Testing Loop, integrated into our existing Supabase + n8n + Vercel stack.

Phase 1: MVP – Engagement Predictor

Train a classification model to predict post success (likes, comments, reach) based on:

Caption text

Image features (model-generated or uploaded)

Platform (LinkedIn, IG, X, etc.)

Posting time

Output: Low / Medium / High performance likelihood score

Help define and refine engagement metrics and training labels

Phase 2: AB Testing Engine

Implement logic to automatically split-test multiple captions or images per post

Track actual performance per variation (time posted, reach, CTR, etc.)

Use Bandit or RL model to optimize future generations

Phase 3: Smart Remix & Auto-Improver

Suggest improved versions of underperforming posts using similarity-based retrieval or model-generated edits

🛠️ Our Stack
Frontend: Next.js

Backend: Supabase (Postgres + Storage), n8n

AI: GPT-4, DALL·E, Klingai, Shotstack

Infra: Vercel, Slack, Stripe

🤝 You’re a Great Fit If You:
Have strong experience in ML classification, NLP, or recommender systems

Have shipped predictive models that were actually used in production

Understand marketing, audience engagement, or content analytics (bonus)

Can work autonomously but love collaborating with technical founders

Are comfortable working async, fast, and on high-impact MVPs

📈 Deliverables
Model training pipeline (cleaned + logged)

First deployed model hosted (we can provide GPU access or API wrapper)

API or webhook endpoint returning prediction scores

Data schema for training labels + performance logs

Slack/Notion async updates + brief Loom walk-throughs

To Apply:
Tell us briefly:

What similar project you’ve done

What model you'd use to predict social media post engagement

Your favorite ML use case in marketing or content

**Comment by abdullahqureshi6** [comment]
Tray Branch 6:49 PM
Hey Muhammad,



Appreciate the thoughtful and well-aligned proposal. Your past work with Authentic, your approach with BERT + CLIP + metadata fusion, and the clear understanding of creative reinforcement make this a strong match for what we're building at Intuity Sync.



To answer your question:



Yes, we're open to MLOps platforms like Weights & Biases or BentoML if they make training and deployment cleaner, especially for managing versioned models and surfacing real-time insights during AB testing. Our priority is velocity + reliability, not reinventing infra.



Quick project context:



We’re tracking post metadata (caption, image URL, time, platform) + plan to log engagement metrics once connected.



Our current tech stack is Supabase, n8n, Vercel, and OpenAI/GPT-4.



We want the model to run asynchronously, likely as a webhook or lightweight API, and provide engagement likelihood scores (Low/Medium/High) per post.



To move forward, could you share:
How you’d break down the Phase 1 delivery (data schema, model training, scoring endpoint)?



Whether you’re comfortable shipping via Notion-based task tracking + async Loom/Slack updates?



How you’d return predictions, JSON format, confidence levels, or something more tailored?



If aligned, we’ll get you scoped and started on a milestone build this week. Excited to collaborate.



—
Tray

**Comment by abdullahqureshi6** [comment]
Hey Tray, Thanks for the clear breakdown everything makes sense and I’m aligned with the async flow (Notion, Loom, Slack) and the stack you’re using.



Here’s how I would approach Phase 1:



* Data schema: Set up structure for storing post metadata and engagement metrics.
* Model training: Use BERT for captions, CLIP for images, and combine them with metadata to train a model that predicts engagement (Low/Medium/High).
* Scoring endpoint: Build a lightweight API or webhook that takes in post info and returns the prediction in JSON format with confidence.



I’m comfortable using tools like Weights & Biases or BentoML if they help with speed and reliability.



Let me know if that sounds good happy to start on the first milestone this week.



Best,
Muhammad.

**Comment by abdullahqureshi6** [comment]
Tray Branch 12:34 PM
Hey Muhammad, this is exactly the approach we were hoping for.



Your fusion of BERT + CLIP + metadata into a Low/Medium/High engagement classifier fits cleanly into the framework we’ve already scoped on our side. Your comfort with async execution (Loom, Slack, Notion) and MLOps tools like W&B or BentoML is ideal, we’re optimizing for fast feedback and modular, reliable delivery.



✅ Let’s Lock Phase 1
Here’s your Milestone 1 – Delivery Scope:



🔧 Phase 1: MVP Predictor



1. Data Schema Setup (You Define, We Match)
You provide a sample schema:



Input: { caption, image_url, platform, post_time }



Output: { label: "Medium", confidence: 0.87 }



We’ll map it to our Supabase tables: prediction_scores, post_variants



2. Mock Dataset (Start with 100–300 examples)
You build or synthesize a test dataset to get training started



We’ll annotate if needed



3. Model Training
BERT embeddings for caption



CLIP/ResNet for basic image vector



Metadata features: platform (one-hot), post time (hour buckets)



Train via XGBoost or simple MLP to start



4. Prediction Endpoint
Lightweight FastAPI/Flask webhook



Accepts: JSON input { caption, image_url, platform, time }



Returns:



{
"engagement_label": "High",
"confidence_score": 0.91
}



5. Delivery Package
🧪 Model + training notebook (Colab or local)



⚙️ API endpoint deployed (or ready-to-deploy script)



📄 README + Loom walkthrough



⏱️ ETA:TBD



🔒 Infra Support from Us:
Supabase anon key



API reference for /predict-score



Backend structure for receiving/storing prediction



Let’s lock this in and get rolling, I’ll create a Notion card for Phase 1 and Slack you the Supabase creds once you confirm timeline.



Appreciate your take on this and if you have a different idea please share I'm all about clarity and precision.



—Tray

**Comment by abdullahqureshi6** [comment]
Tray Branch 1:02 PM
Also, quick scope check before kickoff.



The Phase 1 milestone is scoped for 30 hours or less of focused solo work, covering:
• Caption + metadata feature setup
• XGBoost or LightGBM model
• FastAPI scoring endpoint
• Mock dataset for testing
• Basic docs + Loom walkthrough



Let me know if that’s realistic on your end or if we need to trim/test incrementally. Goal is tight iteration and clean delivery, no overbuild.



Appreciate your precision so far, let’s lock and ship.



—Tray
